addons/stock_delivery/tests/test_carrier_propagation.py

```diff
@@ -126,6 +126,26 @@ def test_carrier_propagation(self):
             ship = pack.move_ids.move_dest_ids.picking_id
             self.assertEqual(self.normal_delivery, ship.carrier_id)
 
+    def test_carrier_propagation_with_all_pull_rules(self):
+        """ Ensure that carrier is propagated through all pull rules when propagate_carrier is enabled,
+        and not propagated for rules where propagate_carrier = False (e.g., the ship rule)."""
+        delivery_route_rules = self.warehouse.delivery_route_id.rule_ids
+        delivery_route_rules[0].location_dest_id = self.rule_pack.location_src_id
+        delivery_route_rules[1].write({'action': 'pull'})
```

Comment:
```suggestion
        delivery_route_rules.action = 'pull'
```
We can avoid a write call.

IMO, if we are `updating multiple fields` at once, using `write() is preferable`. Otherwise, for a `single field assignment`, it’s cleaner and more efficient to assign directly using `=`.